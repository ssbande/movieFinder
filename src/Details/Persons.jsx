import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useParams, withRouter } from 'react-router-dom';
import { Container } from 'semantic-ui-react';
import { connect } from 'react-redux';

import { getDetails } from '../state/actions';
import PersonDetail from './PersonDetails';
import PersonCast from './PersonCast';
import { getDetailInfo } from '../state/selectors';
import constants from '../utils/constants';
import { HistoryProps } from '../utils/propTypes';

const Persons = props => {
  const { artistId } = useParams();
  const { getDetails, detailInfo, history, category } = props;

  useEffect(() => { getDetails(category, artistId) }, [getDetails, category, artistId]);

  const handleCardClick = (id, category) => history.push(`../${constants.categoryRouteMap[category]}/${id}`);

  console.log('detail: ', detailInfo);
  const renderDetails = () => {
    const { combined_credits: { cast } } = detailInfo
    return <Container>
      <PersonDetail detailInfo={detailInfo} />
      <PersonCast cast={cast} handleCardClick={handleCardClick} />
    </Container>
  }

  return <div>
    {!detailInfo
      ? <div>Loading... </div>
      : renderDetails()
    }
  </div>
}

Persons.propTypes = {
  getDetails: PropTypes.func.isRequired,
  history: HistoryProps,
  category: PropTypes.string,
  detailInfo: PropTypes.shape({})
}

const mapDispatchToProps = {
  getDetails: getDetails,
};

const mapState = state => ({
  detailInfo: getDetailInfo(state),
})

export default withRouter(connect(mapState, mapDispatchToProps)(Persons));