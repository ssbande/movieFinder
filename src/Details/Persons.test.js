import React from 'react';
import { shallow } from 'enzyme';

import Persons from './Persons';

const setup = () => ({
  props: {
    getDetails: jest.fn(),
    history: { push: jest.fn() },
    category: 'Persons',
    detailInfo: {
      id: 1001,
      profile_path: 'profile_path',
      title: 'title',
      overview: 'overview',
      genres: [{ name: 'genre1' }, { name: 'genre2' }, { name: 'genre3' }]
    }
  }
})

describe('Persons page tests', () => {
  it('should render correctly', () => {
    const { props } = setup();
    const wrapper = shallow(<Persons {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
})