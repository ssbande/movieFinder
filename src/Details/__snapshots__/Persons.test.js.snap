// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Persons page tests should render correctly 1`] = `ShallowWrapper {}`;
